prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -1447,12 +1447,12 @@ fuse depth = go
 data SimpleDocStream ann =
       SFail
     | SEmpty
-    | SChar Char (SimpleDocStream ann)
+    | SChar !Char (SimpleDocStream ann)
 
     -- | Some layout algorithms use the Since the frequently used 'T.length' of
     -- the 'Text', which scales linearly with its length, we cache it in this
     -- constructor.
-    | SText !Int Text (SimpleDocStream ann)
+    | SText !Int !Text (SimpleDocStream ann)
```

Comment:
Not sure whether this is such a great idea… For example in the layout algorithms we only care about the length.

Reply:
Actually `Doc`'s `Text` constructor is strict in this field too, so it should actually be ok.